activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -98,16 +114,16 @@ def travel_to(date_or_time)
         else
           now = date_or_time.to_time.change(usec: 0)
         end
+        to_date_value = now.to_date
+        to_datetime_value = now.to_datetime
 
-        simple_stubs.stub_object(Time, :now, now)
-        simple_stubs.stub_object(Date, :today, now.to_date)
-        simple_stubs.stub_object(DateTime, :now, now.to_datetime)
+        stub_unstub_date_time_objects(:stub_object, now, to_date_value, to_datetime_value)
 
         if block_given?
           begin
             yield
           ensure
-            travel_back
+            stub_unstub_date_time_objects(:unstub_to_previous_stub_state, now, to_date_value, to_datetime_value)
```

Comment:
With `pop_stub` this could be:

```ruby
simple_stubs.pop_stub(Time, :now)
simple_stubs.pop_stub(Date, :today)
simple_stubs.pop_stub(DateTime, :now)
```

And we could kill `stub_unstub_date_time_objects` and the date and datetime value vars (since stubs remember their `return_value` we can pull it back it out from them).